test/unit/DetectChanges/BCBreak/ClassBased/PropertyChangedTest.php

```diff
@@ -80,4 +80,91 @@ class TheClass {
             )
         );
     }
+
+    /**
+     * @dataProvider finalClassProvider
+     */
+    public function testWillIgnoreChangeInFinalClass($fromLocator, $toLocator) : void
+    {
+
+
+        $comparator = $this->createMock(PropertyBased::class);
+
+        $comparator
+            ->expects(self::exactly(2))
+            ->method('__invoke')
+            ->willReturnCallback(static function (ReflectionProperty $from, ReflectionProperty $to) : Changes {
+                $propertyName = $from->getName();
+
+                self::assertSame($propertyName, $to->getName());
+
+                return Changes::fromList(Change::added($propertyName, true));
+            });
+
+        Assertion::assertChangesEqual(
+            Changes::empty(),
+            (new PropertyChanged($comparator))->__invoke(
+                (new ClassReflector($fromLocator))->reflect('TheClass'),
+                (new ClassReflector($toLocator))->reflect('TheClass')
+            )
+        );
+    }
+
+    public function finalClassProvider() : array
+    {
+        $astLocator = (new BetterReflection())->astLocator();
+
+        return [
+            'final class' => [
+                new StringSourceLocator(
+                    <<<'PHP'
+<?php
+
+final class TheClass {
+    protected $b;
+}
+PHP
+                    ,
+                    $astLocator
+                ),
+                new StringSourceLocator(
+                    <<<'PHP'
+<?php
+final class TheClass {
+    private $b;
+}
+PHP
+                    ,
+                    $astLocator
+                ),
+            ],
+            'final class extending baseclass' => [
+                new StringSourceLocator(
+                    <<<'PHP'
+<?php
+abstract class baseClass {}
+
+final class TheClass extends baseClass {
+    protected $b;
+}
+PHP
+                    ,
+                    $astLocator
+                ),
+                new StringSourceLocator(
+                    <<<'PHP'
+<?php
+abstract class baseClass {}
+
+final class TheClass extends baseClass{
+    private $b;
```

Comment:
Similar here, property `$b` was indeed changed, but not on `baseClass`

Reply:
Because `$b` is not used in `baseClass` it could not be detected as a change.

Reply:
Similar to discussion at https://github.com/Roave/BackwardCompatibilityCheck/pull/202#discussion_r379490933: I think the bug is not in the sources themselves, but rather in the configuration. 

Also see `Roave\BackwardCompatibility\DetectChanges\BCBreak\ClassBased\PropertyRemoved`: that's the one raising this specific warning.